src/Sprites/ImageHeader.cpp

```diff
@@ -54,3 +54,29 @@ void ImageHeader::VerifyIndexedBitCount(uint16_t bitCount)
 		throw std::runtime_error("A bit count of " + std::to_string(bitCount) + " does not support an indexed palette");
 	}
 }
+
+void ImageHeader::Validate()
+{
+	if (headerSize != sizeof(ImageHeader)) {
+		throw std::runtime_error("Image Header must be equal to " + std::to_string(sizeof(ImageHeader)));
+	}
+
+	if (planes != defaultPlanes) {
+		throw std::runtime_error("Planes must be equal to " + std::to_string(defaultPlanes));
```

Comment:
If a user gets this error, they may question why. Is this a format restriction, or is this a support restriction? I think this is due to limited support.

Reply:
Wikipedia says: The number of color planes, must be 1. This is consistent with other sources. When I change the color planes to 0 of a monochrome BMP, MS paint still seems to read it fine.

As far as I know, planes isn't used in any bitmap specification (which makes me wonder what it exists for). So, if we allow it to be a different value, it will not affect anything we are doing. 

It might be an indicator that someone is doing something weird with the file or that the file is damaged? 

I'm happy leaving this in or deleting this check. @DanRStevens, what do you prefer. 

-Brett

Reply:
You mention the Wikipedia article states that value must be 1, but then you have to wonder, why even have that field defined if it can only ever take 1 value? My guess is, it does or did have a use that is now uncommon.

----

If you had a 3 plane image, the red, green, and blue components might be stored in separate arrays, rather than have all 3 color components merged within a single entry within 1 array. Beyond graphics, this has uses for scientific sensors that measure wavelengths outside the visible spectrum, and may have more than 3 bands of data. This would be common for weather satellites, which often capture all kinds of raster data outside the realm of visible light.

Of course if the data represents a raster image outside the visible spectrum, you're going to have to fudge it to display it. Most raster packages are only designed for working with visible image data, so they ignore such concerns.

Though scientific raster data is not typically stored in BMP files. They typically come in alternate compressed raster formats.

----

I think we should definitely have that check. It's pretty clear we don't support multi-plane images.

My concern is about the wording of the message, and how the end user could make sense of it. Right now it would probably make them feel dumb and frustrated. We've effectively passed of responsibility for understanding this problem to the user. That's fine during development, but we need to clean it up. The only time you'd want to leave a message like that in the code is in contexts where the user is more of an expert in their domain than the programmer writing code for them.

Perhaps:
"Image format not supported: Only single plane images are supported, but this image has X planes."

Reply:
I'll update the exception method to what is suggested.

-Brett